src/ott/neural/models/base_solver.py

```diff
@@ -0,0 +1,485 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import abc
+import pathlib
+import types
+from typing import Any, Callable, Dict, Literal, Mapping, Optional, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+
+import optax
+from flax.training import train_state
+
+from ott.geometry import costs, pointcloud
+from ott.problems.linear import linear_problem
+from ott.problems.quadratic import quadratic_problem
+from ott.solvers.linear import sinkhorn
+
+__all__ = ["BaseNeuralSolver", "ResampleMixin", "UnbalancednessMixin"]
+
+
+class BaseNeuralSolver(abc.ABC):
```

Comment:
I don't think there's any need for this class right now, it's just unnecessary abstraction, would remove.